### PR TITLE
Fixed notice if extension image is not found

### LINKED ISSUE #33266: [Issue] Fixed notice if extension image is not found
This issue is automatically created based on existing pull request: magento/magento2#33260: Fixed notice if extension image is not found

---------
### Description (*)
If you add a vimeo video on product, we get an image without extension and you match this video to another product, you will get the following error: Notice : Undefined index: extension in/vendor/magento/framework/File/Uploader on line 802

### Manual testing scenarios (*)
1. Add a vimeo video on product
2. Add the same video on other product

Capture d’écran 2021-06-16 à 14 35 27

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by matthieu2607**
### Description (*)
If you add a vimeo video on product, we get an image without extension and you match this video to another product, you will get the following error: Notice : Undefined index: extension in/vendor/magento/framework/File/Uploader on line 802

### Manual testing scenarios (*)
1. Add a vimeo video on product
2. Add the same video on other product

Capture d’écran 2021-06-16 à 14 35 27

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#33266: Fixed notice if extension image is not found

### CODE REVIEW

**Comment by Den4ik** [comment]
@magento create issue

**Comment by matthieu-barbe** [comment]
@magento run all tests

File: lib/internal/Magento/Framework/File/Uploader.php [hunk 1]
```diff
@@ -803,7 +803,7 @@ public static function getNewFileName($destinationFile)
         $fileInfo = pathinfo($destinationFile);
         $index = 1;
         while ($fileExists($fileInfo['dirname'] . '/' . $fileInfo['basename'])) {
-            $fileInfo['basename'] = $fileInfo['filename'] . '_' . $index++ . '.' . $fileInfo['extension'];
+            $fileInfo['basename'] = (isset($fileInfo['extension'])) ? $fileInfo['filename'] . '_' . $index++ . '.' . $fileInfo['extension'] : $fileInfo['filename'] . '_' . $index++;
```
Comment (Den4ik) [inline lib/internal/Magento/Framework/File/Uploader.php]:
```suggestion
            $fileInfo['basename'] = $fileInfo['filename'] . '_' . $index++ . (isset($fileInfo['extension']) ? '.' . $fileInfo['extension'] : '');
```

Comment (Den4ik) [reply lib/internal/Magento/Framework/File/Uploader.php]: I believe that suggested update look little bit cleaner.
@matthieu-barbe what do you think?

**Review by Den4ik — CHANGES_REQUESTED** [review]
Thanks for fix. Please look at my comment

File: lib/internal/Magento/Framework/File/Uploader.php (hunk 1 above)
Comment (matthieu2607) [reply lib/internal/Magento/Framework/File/Uploader.php]: It's perfect ;)

**Comment by matthieu-barbe** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run Integration Tests, Functional Tests CE, Functional Tests B2B, Functional Tests EE

**Comment by hostep** [comment]
This PR can probably be closed now that https://github.com/magento/magento2/commit/73393245962911247bab4c93a6988e0a1d60dc4d landed?

Thanks to @dfelton for [noticing](https://github.com/magento/magento2/issues/33390#issuecomment-906589514).

**Comment by Den4ik** [comment]
@matthieu-barbe thanks for your time spent on the PR

**Pull request closed without merging**